dts/bindings/i3c/st,stm32-i3c.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2022 Meta Platforms, Inc. and its affiliates.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32H5 I3C controller
+
+compatible: "st,stm32-i3c"
+
+include: [i3c-controller.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  interrupts:
+    required: true
+
+  pinctrl-names:
+    required: true
+
+  dmas:
+    description: |
+      Optional DMA channel specifier, required for DMA transactions.
+      For example dmas for TX/RX/TC/RS on I3C
+          dmas = <&gpdma1 0 120 STM32_DMA_PERIPH_RX
+          &gpdma1 1 121 STM32_DMA_PERIPH_TX
+          &gpdma1 2 122 (STM32_DMA_PERIPH_TX | STM32_DMA_MEM_32BITS | STM32_DMA_PERIPH_32BITS)
+          &gpdma1 3 123 (STM32_DMA_PERIPH_RX | STM32_DMA_MEM_32BITS | STM32_DMA_PERIPH_32BITS)>;
+
+      With, in each cell of the dmas specifier:
+      - &gpdma1: dma controller phandle
+      - 0: channel number (0 to Max-Channel minus 1). From 0 to 7 on stm32h5x.
+      - 120: slot number (request which could be given by the GPDMA)
+      - STM32_DMA_PERIPH_RX: channel configuration (only for srce/dest data size, priority)
```

Comment:
Technically this should rather be descried in dma bindings, and it is already, so I 'd suggest to drop.
One of the reason is that this depends on DMA compatible version, which may vary.